model/src/main/kotlin/VcsInfo.kt

```diff
@@ -79,10 +79,10 @@ data class VcsInfo(
         if (this == EMPTY) return other
 
         return VcsInfo(
-            type.takeUnless { it == EMPTY.type } ?: other.type,
-            url.takeUnless { it == EMPTY.url } ?: other.url,
-            revision.takeUnless { it == EMPTY.revision } ?: other.revision,
-            path.takeUnless { it == EMPTY.path } ?: other.path
+            type = type.takeUnless { it == EMPTY.type } ?: other.type,
```

Comment:
Nit: The commit message could provide a rationale... is readability really and issue here, as each line already has the field name?